ruby/README.md

```diff
@@ -51,6 +52,17 @@ puts MyTestMessage.encode_json(mymessage)
 Installation from Source (Building Gem)
 ---------------------------------------
 
+Protocol Buffers uses the [ffi](https://github.com/ffi/ffi) gem to provide a
+new unified C-based implementation across Ruby interpreters based on
+[UPB](https://github.com/protocolbuffers/upb). For now, use of the FFI
+implementation is opt-in. If any of the following are true, the traditional
+platform-native implementations (MRI-ruby based on CRuby, Java based on JRuby)
+are used instead of the new FFI-based implementation:
+1. `ffi` and `ffi-compiler` gems are not installed
+2. `PROTOCOL_BUFFERS_RUBY_IMPLEMENTATION` environment variable has a value
```

Comment:
Does this environment variable take effect at build time, runtime, or both?

Reply:
Just at runtime.

Reply:
The build time flag (for Bazel only) is `//ruby:ffi=enabled`.

Reply:
If the flag doesn't take effect at build time, it seems somewhat confusing to document it in a section about building gems.

What happens if you build a source gem with Rake instead of Bazel? Do FFI libraries get built?

Reply:
`gem build` includes FFI support unconditionally.

Reply:
The reason it is discussed here despite not being a build time variable is that I've added references to it in the test examples below. I'm attempting to reinforce the idea that just because it is built with FFI support doesn't mean that it is used all of the time.